### PR TITLE
[UITesting, Catalyst] Fixed DragAndDropUITests to be more reliable

### LINKED ISSUE #24914: [UITesting, Catalyst] Fix DragAndDropUITests to be more reliable
### Description

As an example DragAndDropBetweenLayouts fails a good amount on catalyst. We should make the test a bit more reliable if possible.

It looks like it's typically failing in these two spots

1) 
```
   at NUnit.Framework.Legacy.ClassicAssert.True(Boolean condition)
   at Microsoft.Maui.TestCases.Tests.DragAndDropUITests.DragAndDropBetweenLayouts() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs:line 72

1)    at NUnit.Framework.Legacy.ClassicAssert.True(Boolean condition)
   at Microsoft.Maui.TestCases.Tests.DragAndDropUITests.DragAndDropBetweenLayouts() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs:line 72
```

image

2) 

```
 at NUnit.Framework.Legacy.ClassicAssert.True(Boolean condition)
   at Microsoft.Maui.TestCases.Tests.DragAndDropUITests.DragAndDropBetweenLayouts() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs:line 74

1)    at NUnit.Framework.Legacy.ClassicAssert.True(Boolean condition)
   at Microsoft.Maui.TestCases.Tests.DragAndDropUITests.DragAndDropBetweenLayouts() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs:line 74
```

Here's an image of the test when it failed

image

### Thoughts

Instead of just running everything and then testing the string we could perform each operation and then test that that operation completed. 

I've disabled this suite of tests on catalyst [here](https://github.com/dotnet/maui/pull/25039)

### PR DESCRIPTION
**Pull request by NirmalKumarYuvaraj**
### Description of changes
Developed separate test cases for the DragAndDropBetweenLayouts functionality, each covering individual drag-and-drop operations.

#### Validated the behaviour in the following platforms
 
- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Issues Fixed
Fixes #24914

### CODE REVIEW

**Review by jsuarezruiz — APPROVED** [review]

**Review by jfversluis — DISMISSED** [review]

**Review by rmarinho — DISMISSED** [review]

**Review by PureWeen — DISMISSED** [review]
DropEventCoordinates is failing

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]
Failing test on Windows:
image

**Comment by Shalini-Ashokan** [comment]
@PureWeen , @jsuarezruiz 
It seems the failure is unrelated to the recent changes and is occurring in the helper method. I've added a condition to return null if drag coordinates are unavailable.

File: src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs [hunk 1]
```diff
@@ -61,19 +93,61 @@ public void DragAndDropBetweenLayouts()
 			App.WaitForElement("Red");
 			App.WaitForElement("Green");
 			App.DragAndDrop("Red", "Green");
-			App.WaitForElement("DragEventsLabel");
 
-			var textAfterDrag = App.FindElement("DragEventsLabel").GetText();
-			if (string.IsNullOrEmpty(textAfterDrag))
+			App.WaitForElement("DragStartEventsLabel");
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs:97]: The tests are failing on macOS.

```
DragAndDropBetweenLayouts
Assert.That(rainbowColorText, Is.EqualTo("RainbowColorsAdd:Red"))
Expected string length 20 but was 18. Strings differ at index 17.
Expected: "RainbowColorsAdd:Red"
But was:  "RainbowColorsAdd: "
```

image

**Review by jsuarezruiz — CHANGES_REQUESTED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs (hunk 1 above)
Comment (sheiksyedm) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/DragAndDropUITests.cs:97]: @jfversluis  Windows and Mac test cases are randomly failing on CI machines, even though they work properly in local machines. The programmatic drag-and-drop operation using App.DragAndDrop(source, target) is not functioning correctly on Windows and Mac in the CI environment. It seems that the pointer offset coordinates on these devices differ in the CI. Do you have any suggestions for resolving this kind of issue?

**Review by jsuarezruiz — APPROVED** [review]

**Comment by PureWeen** [comment]
- failing test not related

**Pull request merged**